dts/bindings/gpio/ambiq-header.yaml

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    GPIO pins exposed on Ambiq Apollo4p EVB headers
```

Comment:
This seems very specific to that one board. Do you plan to add any shields matching this header, or why is the binding here?

Reply:
As mentioned in https://github.com/zephyrproject-rtos/zephyr/pull/61157#issuecomment-1719012866, we plan to add a new shield in https://github.com/zephyrproject-rtos/zephyr/pull/61895. We will reopen this PR after merging this one.